### PR TITLE
feat: add pdb for sbx/dev/test envs + logic

### PR DESCRIPTION
**Pull request by Glenn-Terjesen**

### CODE REVIEW

**Comment by AlexanderBrevig** [comment]
We should add some tests to prove the new functionality and make sure we don't accidentally regress.

**Review by AlexanderBrevig — CHANGES_REQUESTED** [review]
needs some unit tests to prove the change

**Comment by Glenn-Terjesen** [comment]
@AlexanderBrevig added multiple tests, please take a look :-)

**Review by AlexanderBrevig — CHANGES_REQUESTED** [review]
Consider following the established pattern that when keys are redundant between the `deployment` stanza and `container`, they are prioritized for `deployment`.

`pdb.minAvailable > deployment.minAvailable > container.minAvailable`

File: charts/common/tests/pdb_test.yaml [hunk 1]
```diff
@@ -76,13 +92,35 @@ tests:
...
       - equal:
           path: spec.minAvailable
           value: "25%"
+  - it: must not use pdb if Replicas is set to 1
+    set:
+      <<: *values
+      env: prd
+      container:
+        image: some
+        Replicas: 1
```
Comment (AlexanderBrevig) [inline charts/common/tests/pdb_test.yaml:109]: should be lower case

**Review by AlexanderBrevig — CHANGES_REQUESTED** [review]

**Review by AlexanderBrevig — APPROVED** [review]
🎊

**Pull request merged**